pennylane/devices/tests/test_gates_with_expval.py

```diff
@@ -0,0 +1,364 @@
+                        [
+                            [-1j / sqrt(2), 1j / sqrt(2)],
+                            [1j / sqrt(2), 1j / sqrt(2)],
+                        ]
+                    )
+                ],
+                0,
+            ),
+        ],
+    )
+    def test_supported_gate_single_wire_with_parameters(
+        self, device, tol, name, par, expected_output
+    ):
+        """Tests supported gates that act on a single wire that are parameterized"""
+
+        n_wires = 1
+        dev = device(n_wires)
+        op = getattr(qml.ops, name)
+
+        @qml.qnode(dev)
+        def circuit():
+            op(*par, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        assert np.isclose(circuit(), expected_output, atol=tol(dev.shots))
+
+    # This test is ran against the state 1/2|00>+sqrt(3)/2|11> with two Z expvals
+    @pytest.mark.parametrize(
+        "name,par,expected_output",
+        [
+            ("CRX", [0], [-1 / 2, -1 / 2]),
+            ("CRX", [-pi], [-1 / 2, 1]),
+            ("CRX", [pi / 2], [-1 / 2, 1 / 4]),
+            ("CRY", [0], [-1 / 2, -1 / 2]),
+            ("CRY", [-pi], [-1 / 2, 1]),
+            ("CRY", [pi / 2], [-1 / 2, 1 / 4]),
+            ("CRZ", [0], [-1 / 2, -1 / 2]),
+            ("CRZ", [-pi], [-1 / 2, -1 / 2]),
+            ("CRZ", [pi / 2], [-1 / 2, -1 / 2]),
+            ("MultiRZ", [0], [-1 / 2, -1 / 2]),
+            ("MultiRZ", [-pi], [-1 / 2, -1 / 2]),
+            ("MultiRZ", [pi / 2], [-1 / 2, -1 / 2]),
+            ("CRot", [pi / 2, 0, 0], [-1 / 2, -1 / 2]),
+            ("CRot", [0, pi / 2, 0], [-1 / 2, 1 / 4]),
+            ("CRot", [0, 0, pi / 2], [-1 / 2, -1 / 2]),
+            ("CRot", [pi / 2, 0, -pi], [-1 / 2, -1 / 2]),
+            ("CRot", [0, pi / 2, -pi], [-1 / 2, 1 / 4]),
+            ("CRot", [-pi, 0, pi / 2], [-1 / 2, -1 / 2]),
+            (
+                "QubitUnitary",
+                [
+                    np.array(
+                        [
+                            [1, 0, 0, 0],
+                            [0, 1 / sqrt(2), 1 / sqrt(2), 0],
+                            [0, 1 / sqrt(2), -1 / sqrt(2), 0],
+                            [0, 0, 0, 1],
+                        ]
+                    )
+                ],
+                [-1 / 2, -1 / 2],
+            ),
+            (
+                "QubitUnitary",
+                [
+                    np.array(
+                        [
+                            [-1, 0, 0, 0],
+                            [0, 1 / sqrt(2), 1 / sqrt(2), 0],
+                            [0, 1 / sqrt(2), -1 / sqrt(2), 0],
+                            [0, 0, 0, -1],
+                        ]
+                    )
+                ],
+                [-1 / 2, -1 / 2],
+            ),
+        ],
+    )
+    def test_supported_gate_two_wires_with_parameters(
+        self, device, tol, name, par, expected_output
+    ):
+        """Tests supported gates that act on two wires wires that are parameterized"""
```

Comment:
Check the docsting, I am not sure if I understand correctly, are the gates parametrised?